jre task has missing dependency on jar task

Running Gradle with the `--parallel` flag can expose an undeclared dependency of the `jre` task on the `jar` task.

I've made an [example project](https://github.com/markslater/runtime-query) that reproduces the issue (on my machine... given that it involves parallelism, different hardware may produce different results).

In summary, my `build.gradle.kts` is:
```kts
plugins {
    kotlin("jvm") version "1.5.30"
    id("org.beryx.runtime") version "1.12.5"
    id("org.jlleitschuh.gradle.ktlint") version "10.1.0"
}

java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(16))
    }
}

application {
    mainClass.set("com.example.Example")
}

runtime {
    options.set(listOf("--strip-debug", "--compress", "2", "--no-header-files", "--no-man-pages"))
}
```
If I run `./gradlew clean runtime` everything works, but if I run `./gradlew --parallel clean runtime`, it fails:
```
> Task :jre FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':jre'.
> File or directory not found: /home/mark/IdeaProjects/runtime-query/build/libs/example.jar
```
I note that [`JreTask.groovy`](https://github.com/beryx/badass-runtime-plugin/blob/v1.12.5/src/main/groovy/org/beryx/runtime/JreTask.groovy#L64) doesn't declare any dependencies, whereas `SuggestModulesTask.groovy` [does](https://github.com/beryx/badass-runtime-plugin/blob/v1.12.5/src/main/groovy/org/beryx/runtime/SuggestModulesTask.groovy#L32), so my guess is that in single threaded mode, it works due to implicit task ordering (maybe order of task declaration?).  Note that applying the `ktlint` plugin was necessary to reproduce the issue on my machine... I guess because it adds pre-compile tasks, delaying the production of the jar.

I'd be happy to submit a PR if it would be helpful.

--- a/src/main/groovy/org/beryx/runtime/JreTask.groovy
+++ b/src/main/groovy/org/beryx/runtime/JreTask.groovy
@@ -62,6 +62,7 @@ class JreTask extends BaseTask {
     }
 
     JreTask() {
+        dependsOn('jar')
         description = 'Creates a custom java runtime image with jlink'
     }
 

